posts/stargate.md

```diff
@@ -0,0 +1,246 @@
+- CQL service (**9042**)
+
+Make sure that the container was started without any errors before continuing this blog.
+
+![Output](https://raw.githubusercontent.com/Falydoor/blog-usa/stargate/images/2021/01/stargate-3.png)
+
+# Inserting data using the REST API
+
+## Overview
+
+The rest API is secured using a classic token stored in the header `X-Cassandra-Token`. The curl request below will generate a token and then you just have to store it in the environment variable `AUTH_TOKEN`:
+
+```bash
+curl -L -X POST 'http://localhost:8081/v1/auth' \
+  -H 'Content-Type: application/json' \
+  --data-raw '{
+    "username": "cassandra",
+    "password": "cassandra"
+}'
+
+# paste value of authToken
+export AUTH_TOKEN=
+```
+
+Make sure that everything is working by checking that the keyspace `system` exists:
+
+```bash
+curl -L -X GET 'localhost:8082/v2/schemas/keyspaces/system' \
+-H "X-Cassandra-Token: $AUTH_TOKEN" \
+-H 'Content-Type: application/json'
+```
+
+For more details about the REST API, please refer to the [documentation](https://stargate.io/docs/stargate/1.0/developers-guide/rest-using.html).
+
+## Data insertion
+
+In order to easily generate data, I wrote a small Node.js script that will generate "fake" vehicles using the library [Faker.js](https://github.com/marak/Faker.js/) and make REST requests using [Axios](https://github.com/axios/axios).
+
+Start by cloning my [repository](https://github.com/Falydoor/blog-stargate) and then run `npm install` to install the required libraries.
+
+Run the script with `node server.js` to create a keyspace `blog` and insert 50 different vehicles in the table `vehicle`. Your console should output the type and manufacturer of all inserted vehicles:
+
+![Output](https://raw.githubusercontent.com/Falydoor/blog-usa/stargate/images/2021/01/stargate-1.png)
+
+Here is the table definition of `vehicle`:
+
+```json
+{
+  "data": {
+    "name": "vehicle",
+    "keyspace": "blog",
+    "columnDefinitions": [
+      {
+        "name": "manufacturer",
+        "typeDefinition": "varchar",
+        "static": false
+      },
+      {
+        "name": "type",
+        "typeDefinition": "varchar",
+        "static": false
+      },
+      {
+        "name": "color",
+        "typeDefinition": "varchar",
+        "static": false
+      },
+      {
+        "name": "model",
+        "typeDefinition": "varchar",
+        "static": false
+      },
+      {
+        "name": "vin",
+        "typeDefinition": "varchar",
+        "static": false
+      }
+    ],
+    "primaryKey": {
+      "partitionKey": [
+        "manufacturer"
+      ],
+      "clusteringKey": [
+        "type"
+      ]
+    },
+    "tableOptions": {
+      "defaultTimeToLive": 0,
+      "clusteringExpression": [
+        {
+          "order": "ASC",
+          "column": "type"
+        }
+      ]
+    }
+  }
+}
+```
+
+The partition key is `manufacturer` and the clustering key is `type` so we can get all vehicles by manufacturer and sort them by type.
```

Comment:
Is that a "clustering key" or "sort key"?

Reply:
Clustering key yeah